pkg/latte/intent_reset_password.go

```diff
@@ -30,28 +30,42 @@ func (*IntentResetPassword) JSONSchema() *validation.SimpleSchema {
 }
 
 func (*IntentResetPassword) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
-	if len(w.Nodes) == 0 {
+	switch len(w.Nodes) {
+	case 0:
 		return []workflow.Input{
-			&InputResetPasswordByCode{},
+			&InputTakeCode{},
+		}, nil
+	case 1:
+		return []workflow.Input{
+			&InputTakePassword{},
 		}, nil
 	}
 	return nil, workflow.ErrEOF
 }
 
 func (i *IntentResetPassword) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
-	var inputResetPasswordByCode inputResetPasswordByCode
-
-	switch {
-	case workflow.AsInput(input, &inputResetPasswordByCode):
-		node := NodeDoResetPasswordByCode{
-			Code:        inputResetPasswordByCode.GetCode(),
-			NewPassword: inputResetPasswordByCode.GetNewPassword(),
+	switch len(w.Nodes) {
+	case 0:
+		var inputTakeCode inputTakeCode
+		if workflow.AsInput(input, &inputTakeCode) {
+			node := NodeValidateResetPasswordCode{
```

Comment:
Instead of validating the code in effect, validate the code here and produce NodeValidatedResetPasswordCode.

NodeValidatedResetPasswordCode represents the code is proven to be valid.